src/Storages/WindowView/StorageWindowView.cpp

```diff
@@ -1020,7 +1020,7 @@ void StorageWindowView::threadFuncFireProc()
 
     std::lock_guard lock(fire_signal_mutex);
     /// TODO: consider using time_t instead (for every timestamp in this class)
-    UInt32 timestamp_now = static_cast<UInt32>(std::time(nullptr));
+    UInt32 timestamp_now = static_cast<UInt32>(Poco::Timestamp().epochMicroseconds() / 1000000);
```

Comment:
but there are many other usages of std::time(nullptr) in code of WindowView, should we then change it everywhere for consistency?
Also does it really make sense if as you say the difference can be just a few milliseconds, but here we use seconds, so the difference can be negligible?

Reply:
The problem is that `time()` is less a few milliseconds - the result is that on L1025 `next_fire_signal` > `timestamp_now`, then `next_fire_signal` stays the same, and on L1049 we reschedule this task for 0 milliseconds - it fires immediately and this function is called again. And again. And again - possibly few dozens times until `time()` catches up with the scheduler timer.
Regarding other places - probably, I'll check...